### PR TITLE
Support 'where' argument in jax.numpy reductions

### PR DESCRIPTION
**Pull request by jakevdp**

### CODE REVIEW

**Review by froystig — APPROVED** [review]
Nice!

**Comment by domenicjdonato** [comment]
I think this change breaks jax.nn.softmax(tensor, axis=-1). Can you please double check this?

**Comment by jakevdp** [comment]
> I think this change breaks jax.nn.softmax(tensor, axis=-1). Can you please double check this?

Perhaps... could you open an issue with an example of the error?
